src/App.tsx

```diff
@@ -14,11 +16,123 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  goods: string[]
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goods: ['Jam'],
+  };
+
+  saveGood = (selectedGood: string) => {
+    this.setState((state) => ({
+      goods: [...state.goods, selectedGood],
+    }));
+  };
+
+  clearState = () => {
+    this.setState({ goods: [] });
+  };
+
+  removeGood = (selectedGood: string) => {
+    this.setState((state) => (
+      { goods: state.goods.filter(good => good !== selectedGood) }
+    ));
+  };
+
+  goodsTitle = () => {
+    const { goods } = this.state;
+    let title = '';
+
+    switch (goods.length) {
+      case 0: {
+        title = 'No goods selected';
+        break;
```

Comment:
Here you can just return from method needed string, it helps you don't write break statement.